cfme/tests/services/test_service_rbac.py

```diff
@@ -0,0 +1,143 @@
+import fauxfactory
+import pytest
+
+from cfme.base.credential import Credential
+from cfme.utils.blockers import BZ
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.services.service_catalogs import ServiceCatalogs
+from cfme.utils import error
+
+pytestmark = [
+    pytest.mark.usefixtures('uses_infra_providers'),
+    pytest.mark.usefixtures('setup_provider_modscope'),
+    pytest.mark.provider([VMwareProvider], scope="module")]
+
+
+def new_credential():
+    if BZ(1487199, forced_streams=['5.8']).blocks:
+        return Credential(principal='uid{}'.format(fauxfactory.gen_alphanumeric().lower()),
+                          secret='redhat')
+    else:
+        return Credential(principal='uid{}'.format(fauxfactory.gen_alphanumeric()),
+                          secret='redhat')
+
+
+def new_role(appliance, product_features):
+    collection = appliance.collections.roles
+    return collection.create(name='role_{}'.format(fauxfactory.gen_alphanumeric()),
+                             vm_restriction=None, product_features=product_features)
+
+
+def new_group(appliance, role):
+    collection = appliance.collections.groups
+    return collection.create(description='group_{}'.format(fauxfactory.gen_alphanumeric()),
+                             role=role, tenant="My Company")
+
+
+def new_user(appliance, group):
+    collection = appliance.collections.users
+    return collection.create(name='user_{}'.format(fauxfactory.gen_alphanumeric()),
+                             credential=new_credential(),
+                             email='xyz@redhat.com',
+                             groups=group,
+                             cost_center='Workload',
+                             value_assign='Database')
+
+
+@pytest.yield_fixture(scope="module")
+def role_user_group(appliance):
+    role = new_role(
+        appliance=appliance,
+        product_features=[(['Everything'], False)] + [(['Everything', k], True)
+                          for k in ['Access Rules for all Virtual Machines']])
+    group = new_group(appliance=appliance, role=role.name)
+    user = new_user(appliance=appliance, group=group)
+    yield (role, user)
+
+
+@pytest.yield_fixture(scope='function')
+def _myservice(appliance, catalog_item):
+    vm_name = catalog_item.provisioning_data["catalog"]["vm_name"] + '_0001'
+    catalog_item.create()
+    service_catalogs = ServiceCatalogs(appliance, catalog_item.catalog, catalog_item.name)
+    service_catalogs.order()
+    yield catalog_item.name, vm_name
+
+
+def test_service_rbac_no_permission(appliance, role_user_group):
+    """ Test service rbac without user permission """
+    role, user = role_user_group
+    error_message = "The user's role is not authorized for any access, " \
```

Comment:
error_message = ("The user's role is not authorized for any access, "
                             "please contact the administrator!")